api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -146,19 +205,21 @@ private Order saveOrder(Order order, OrderContext orderContext, boolean isRetros
 		if (isDrugOrder) {
 			((DrugOrder) order).setAutoExpireDateBasedOnDuration();
 		}
-		
 		if (concept == null) {
 			throw new MissingRequiredPropertyException("Order.concept.required");
 		}
-		
+
+	}
+
+	private void ensureOrderTypeIsSet(Order order, OrderContext orderContext) {
 		Order previousOrder = order.getPreviousOrder();
 		if (order.getOrderType() == null) {
```

Comment:
@Nuwantha we can simplify this.
turn this into a guard clause like
```java
if (order.getOrderType() != null) {
  return;
}
Order previousOrder = order.getPreviousOrder();
...
```

this way you remove one level of nesting in this method and you signal the reader that ok if the `orderType` is already set, nothing to do here, return.